js/components/configuration.js

```diff
@@ -1,21 +1,89 @@
-define(['knockout', 'text!./configuration.html', 'appConfig', 'webapi/AuthAPI', 'atlas-state', 'access-denied'], function (ko, view, config, authApi, sharedState) {
+define(['knockout', 'text!./configuration.html', 'appConfig', 'webapi/AuthAPI', 'webapi/SourceAPI', 'atlas-state', 'access-denied'], function (ko, view, config, authApi, sourceApi, sharedState) {
 	function configuration(params) {
 		var self = this;
 		self.config = config;
 		self.api = config.api;
 		self.sharedState = sharedState;
+    self.isInProgress = ko.observable(false);
+		self.sources = ko.observableArray(config.api.sources);
+		self.priorityOptions = [
+      {name: 'Current Session', id: 'session'},
+      {name: 'Whole Application', id: 'application'},
+    ];
 
 		self.isAuthenticated = authApi.isAuthenticated;
 		self.hasAccess = ko.pureComputed(function () {
 			return (config.userAuthenticationEnabled && self.isAuthenticated() && authApi.isPermittedEditConfiguration()) || !config.userAuthenticationEnabled;
 		});
 		self.canReadRoles = ko.pureComputed(function () {
 			return self.isAuthenticated() && authApi.isPermittedReadRoles();
-		})
+		});
+		self.canCreateSource = ko.pureComputed(function () {
+      return (config.userAuthenticationEnabled && self.isAuthenticated() && authApi.isPermittedCreateSource()) || !config.userAuthenticationEnabled;
+    });
+    self.canChangePriority = ko.pureComputed(function () {
+      return (config.userAuthenticationEnabled && self.isAuthenticated() && authApi.isPermittedEditSourcePriority()) || !config.userAuthenticationEnabled;
+    });
+    self.canReadSource = function(source) {
+      return (config.userAuthenticationEnabled && self.isAuthenticated()
+        && authApi.isPermittedReadSource(source.sourceKey)) || !config.userAuthenticationEnabled;
+    };
 		self.clearLocalStorageCache = function () {
 			localStorage.clear();
 			alert("Local Storage has been cleared.  Please refresh the page to reload configuration information.")
-		}
+		};
+
+		self.newSource = function () {
+			document.location = "#/source/new";
+    };
+
+		self.selectSource = function(source) {
+			document.location = "#/source/" + source.sourceId;
+		};
+
+    function updateSourceDaimonPriority(sourceKey, daimonType) {
+      if (self.sharedState.priorityScope() !== 'application') {
+        return;
+      }
+      self.isInProgress(true);
+      $.ajax({
+        url: config.api.url + 'source/' + sourceKey + '/daimons/' + daimonType + '/set-priority',
+        timeout: 20000,
+        method: 'POST',
+        contentType: 'application/json',
+        success: function () {
+          sourceApi.initSourcesConfig();
+        },
+        error: function (err) {
+          self.isInProgress(false);
+          alert('Failed to update priority source daimon');
+        },
+        complete: function () {
+          self.isInProgress(false);
```

Comment:
From jQuery doc - "A function to be called when the request finishes (after success and error callbacks are executed)"

So no need to call same line in error callback